views/doc.list.html.twig

```diff
@@ -3,36 +3,61 @@
 {% import _self as macros %}
 
 {% block content %}
-    <div id="searchbar" class="clearfix">
-        <input id="docsearch" type="text" placeholder="Search documentation">
-    </div>
-
-    <h1>Book</h1>
-    <ul>
-    {% for filename, data in book %}
-        {{ macros.renderFile(filename, data) }}
-    {% endfor %}
-    </ul>
-    <h1>Articles</h1>
-    <ul>
-    {% for filename, data in articles %}
-        {{ macros.renderFile(filename, data) }}
-    {% endfor %}
-    </ul>
-    <h1>FAQs</h1>
-    <ul>
-    {% for filename, data in faqs %}
-        {{ macros.renderFile(filename, data) }}
-    {% endfor %}
-    </ul>
-
-    <h1>API Docs</h1>
-    <p>If you need to work with composer as a library you may also want to check out the <a href="/apidoc/master/index.html">API Docs</a></p>
+    <article role="article">
+
+        <div id="searchbar" class="clearfix">
+            <input id="docsearch" type="search" placeholder="Type here to search documentation..." role="searchbox"
+                   title="Enter text here and submit to search the documentation"/>
+        </div>
+
+        <h1 role="heading">Book</h1>
+        <ul>
+            {% for filename, data in book %}
+                {# Empty sections here would make no sense #}
+                {% if data is not empty and filename is not empty %}
```

Comment:
why are you adding that condition ? AFAICT, this condition will always be true, due to the way the variables are built.

Reply:
Defensive programming. Always looks useless. Better safe than sorry, right? ;)

Reply:
well, we control both sides of the code here. This only makes the template less readable IMO.

Reply:
Well, according to controllers.php@51 they could be empty or "malformed". But OK, if you prefer to remove those free double-check controls, let's trim them out. :)